lib-impl/src/main/java/org/ergoplatform/appkit/impl/Eip29AttachmentBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.ergoplatform.appkit.impl;
+
+import static org.ergoplatform.appkit.impl.Eip4TokenBuilder.getSerializedErgoValueForRegister;
+
+import org.ergoplatform.appkit.Eip29Attachment;
+import org.ergoplatform.appkit.ErgoValue;
+import org.ergoplatform.appkit.TransactionBox;
+import org.ergoplatform.explorer.client.model.AdditionalRegisters;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import scala.Tuple2;
+
+public class Eip29AttachmentBuilder {
+
+    /**
+     * @return EIP-29 compliant attachment built from serialized ergo value
+     */
+    public static Eip29Attachment buildFromHexEncodedErgoValue(String hexEncodedErgoValue) {
+        return Eip29Attachment.createFromErgoValue(ErgoValue.fromHex(hexEncodedErgoValue));
+    }
+
+    /**
+     * @return EIP-29 compliant attachment, if found. null otherwise
+     */
+    @Nullable
+    public static Eip29Attachment buildFromAdditionalRegisters(@Nonnull AdditionalRegisters registers) {
+        try {
+            return buildFromHexEncodedErgoValue(getSerializedErgoValueForRegister(registers, "R9"));
+        } catch (Throwable t) {
+            return null;
+        }
+    }
+
+    /**
+     * @return EIP-29 compliant attachment, if found. null otherwise
+     */
+    @Nullable
+    public static Eip29Attachment buildFromTransactionBox(@Nonnull TransactionBox transactionBox) {
+        List<ErgoValue<?>> boxRegisters = transactionBox.getRegisters();
+        try {
+            return Eip29Attachment.createFromErgoValue(boxRegisters.get(5));
```

Comment:
Better to use the source metadata Instead of introducing the magic constant  https://github.com/ScorexFoundation/sigmastate-interpreter/blob/0fa0448ee41012891a28eaca92bcc768234bd08f/sigmastate/src/main/scala/org/ergoplatform/ErgoBox.scala#L155